From 68948daf497f3ab05c06725cda26f61f20a222e2 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Tue, 23 Jun 2015 23:35:30 -0400
Subject: [PATCH] EMC Specific User API's

For quick access to our server user data object and user id
---
 .../java/org/bukkit/craftbukkit/entity/CraftPlayer.java  | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index 4f31ae6356..292cf8c81d 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -77,6 +77,22 @@ import javax.annotation.Nullable;
 
 @DelegateDeserialization(CraftOfflinePlayer.class)
 public class CraftPlayer extends CraftHumanEntity implements Player {
+    // EMC start
+    private Object user;
+    private long userId;
+    public long getUserId() {
+        return userId;
+    }
+    public void setUserId(long userId) {
+        this.userId = userId;
+    }
+    public <T> T getUser() {
+        return (T) user;
+    }
+    public void setUser(Object user) {
+        this.user = user;
+    }
+    // EMC end
     private long firstPlayed = 0;
     private long lastPlayed = 0;
     private boolean hasPlayedBefore = false;
-- 
2.15.1

